### ISSUE TITLE
Please help, can't install or get this extension to show up in VLC.

### ISSUE THREAD

**Issue by gkennedy88** [Anndrey24/VLC_Speed_Controller#5]
@Anndrey24 
Hello,

I read through the instructions on the main page, it says to "put the vlc_speed_controller.lua file in" then goes on to list the following two directories for windows pc's, "all users\ProgramFiles\VideoLAN\VLC\lua\extensions", as well as "current user\APPDATA\VLC\lua\extensions".

However when I went to look for the VLC folder under my current user account's appdata folder, there is no such folder located there. But in the appdata folder there is a folder titled "roaming" in which there is a VLC folder, which contains the LUA folder, and inside the LUA folder is the extensions folder, which is also where the VLSub extension I already had installed is located, so I assumed that you accidentally left the "roaming" folder out of the directory line and that it should actually read as "current user\APPDATA\Roaming\VLC\lua\extensions"

I have even tried making the directories directly in the app data folder so that the lua file is in the directory you listed instead of putting it under appdata\roaming\VLC\lua\extensions, however after doing that it still doesn't show up under the view tab. The only extension that shows up is still VLSub. I even went and checked under the Tools tabs Plugin & Extensions Manager, and while the speed extension does show up now, it doesn't show up under the extension tab, its located under the AddOns manager tab.
 
So I have no idea what's going on, or why its not showing up where its supposed to be in VLC at all.

Do you have any idea why its not showing up, or working correctly, or suggestions that would help me to get it to work, or show up correctly?

Looking forward to your response, because I really don't like the standard speed controller inside VLC, that only offers Faster, Faster (Fine), Normal, Slower (Fine), and Slower. it needs more control like i can see in your extension.

**Comment by Anndrey24** [comment]
Hi @gkennedy88,

That is very weird indeed. I'm guessing you have already tried placing it only in the `%ProgramFiles%\VideoLAN\VLC\lua\extensions\` path instead? That's the way I have installed it on my machines so far, the other is just an alternative I found online but never used myself. Still, I've just tried creating the folder in `user\APPDATA\Roaming\vlc\lua\extensions` as you said and the extension shows up in `Toolbar > View` without a problem.

Could you tell me the following so I can maybe recreate the issue locally?
- Windows version (10/11)
- VLC version
- The filename of the extension that you've downloaded so I know the version

Also, it could be worth doing the following and sending me the output you get:
 1. Open the `Toolbar > Tools > Messages`
 2. Set `Verbosity: 2 (debug)` and leave the window open
 3. Open `Toolbar > Tools > Plugins and extensions > Active extensions` 
 4. Click "Reload extensions"
 5. Check the "Messages" window to see the output 

I also hope we can get this working since I agree that the default VLC speed controls are just horrendous.

**Comment by gkennedy88** [comment]
Hello again @Anndrey24 

To answer your first question, yes I did first try using it with it only installed under Program files\videoLAN\VLC\Lua\Extensions\. May I ask why do you type it as %programfiles% because there are no % symbols in the title of the program files folder, it’s just “Program Files”.

Windows version: Windows 10 Professional
VLC Version: 3.0.18 Vetinari
The file that I downloaded is the Lua file that is posted on this site under the code tab for the program, the way the file is titled on here is the exact same way it is titled on my computer, which is - vlc_speed_controller.lua

After removing the folders I created under my local user account, and only having it in the program files section, this is what the Debug Messages displayed:

main debug: looking for addons finder module matching "addons.store.list": 3 candidates
main debug: using addons finder module "fsstorage"
main debug: removing module "fsstorage"
qt debug: ExtensionsDialogProvider is quitting...
main debug: removing module "lua"
main debug: looking for extension module matching "any": 1 candidates
lua debug: Opening Lua Extension module
lua debug: Trying Lua scripts in C:\Users\Shea Kennedy\AppData\Roaming\vlc\lua\extensions
lua debug: Trying Lua scripts in C:\Program Files\VideoLAN\VLC\lua\extensions
lua debug: Trying Lua playlist script C:\Program Files\VideoLAN\VLC\lua\extensions\VLSub.luac
lua debug: Scanning Lua script C:\Program Files\VideoLAN\VLC\lua\extensions\VLSub.luac
lua debug: Script C:\Program Files\VideoLAN\VLC\lua\extensions\VLSub.luac has the following capability flags: 0x5
lua debug: Trying Lua playlist script C:\Program Files\VideoLAN\VLC\lua\extensions\vlc_speed_controller.lua
lua debug: Scanning Lua script C:\Program Files\VideoLAN\VLC\lua\extensions\vlc_speed_controller.lua
lua warning: Error loading script C:\Program Files\VideoLAN\VLC\lua\extensions\vlc_speed_controller.lua: C:\Program Files\VideoLAN\VLC\lua\extensions\vlc_speed_controller.lua:7: unexpected symbol near '
main debug: using extension module "lua"

Hope this helps you figure out what is going on so that I can get it working.

**Comment by Anndrey24** [comment]
The `%Program Files%` is just to denote the Program Files folder in which you've done your installation, be it `Program Files` or `Program Files (x86)`.

I have the same setup but I still cannot reproduce the error. The second to last line from your output would indicate the error coming from some of the comments at the beginning of the file, though I don't see how that could happen. Would you mind opening the extension file in a text editor and removing the first 19 lines, so that the file now starts with the `DEFAULT_CONFIG = {` line, then try the same thing as before and resend the output you get?

**Comment by gkennedy88** [comment]
@Anndrey24 

there is no Defualt_Config = { 
line in the file as far as i can see its just full of hyperlinks to webpages

this is what I see when I open the LUA file in Microsoft word:

Skip to content  
Sign up 
•	Product 
o	
Actions
Automate any workflow 

o	
Packages
Host and manage packages 

o	
Security
Find and fix vulnerabilities 

o	
Codespaces
Instant dev environments 

o	
Copilot
Write better code with AI 

o	
Code review
Manage code changes 

o	
Issues
Plan and track work 

o	
Discussions
Collaborate outside of code 

•	Explore 
o	All features 
o	Documentation 
o	GitHub Skills 
o	Blog 
•	Solutions 
For 
o	Enterprise 
o	Teams 
o	Startups 
o	Education 
By Solution 
o	CI/CD & Automation 
o	DevOps 
o	DevSecOps 
Resources 
o	Customer Stories 
o	White papers, Ebooks, Webinars 
o	Partners 
•	Open Source 
o	
GitHub Sponsors
Fund open source developers 

o	
The ReadME Project
GitHub community articles 

•	Repositories 
o	Topics 
o	Trending 
o	Collections 
•	Pricing 
Search or jump to... 


Sign in 
Sign up 
You signed in with another tab or window. Reload to refresh your session. You signed out in another tab or window. Reload to refresh your session. You switched accounts on another tab or window. Reload to refresh your session. 
Dismiss alert 
{{ message }}
Anndrey24 / VLC_Speed_Controller Public 
•	Notifications 
•	Fork 4 
•	Star 8 
•	Code 
•	Issues 1 
•	Pull requests 0 
•	Actions 
•	Projects 0 
•	Security 
•	Insights 
More 
•	Code 
•	Issues 
•	Pull requests 
•	Actions 
•	Projects 
•	Security 
•	Insights 
Footer
© 2023 GitHub, Inc. 
Footer navigation
•	Terms
•	Privacy
•	Security
•	Status
•	Docs
•	Contact GitHub
•	Pricing
•	API
•	Training
•	Blog
•	About
You can’t perform that action at this time. 

I'm assuming this is not what is supposed to be in there, which must mean I somehow downloaded the wrong file or something?

I don't know how else to save the file. I right clicked on it under the code tab main page, and clicked save link as, and it downloaded it. The only other option I saw was when I clicked on the file instead it took me to a webpage with the option to download the RAW file, but I don't have any programs to open or run RAW files, so I figured that was the wrong way to do it.

**Comment by gkennedy88** [comment]
@Anndrey24 

Well I feel stupid now. I guess saving it the way I did was defiantly the wrong way, I deleted the original file from the program files location, then went back onto the code tab here on this site, and instead of right clicking on the LUA file, I clicked on it like I was clicking on a link, and then went ahead and clicked download RAW file, saved it, and then copied it into the program files folder location, and now it shows up in VLC like its supposed to.

Like I said I feel stupid now. However I honestly thought that it was going to download a .Raw file since it says "Download Raw File". But oh well, thanks for your help, if not for you, I'd have never figured out the problem, much less gotten it to work. 

So thanks!
